src/main/java/com/onlinebookstore/service/impl/UserServiceImpl.java

```diff
@@ -23,6 +25,7 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
             throw new RegistrationException("Can't register user '" + requestDto.getEmail()
                     + "' (already exists)");
         }
+        requestDto.setPassword(passwordEncoder.encode(requestDto.getPassword()));
```

Comment:
```suggestion
        User user = userMapper.toUser(requestDto);
        user.setRoles(.....
        user.setPassword(encoder.encode(requestDto.password()));
        userRepository.save(user);
        return userMapper.toUserResponseDto(user);
```

Where `roleRepository.findByName(Role.RoleName.ROLE_USER)`